### PR TITLE
gh-107028: tiny textual changes in logging docs and docstrings

### PR DESCRIPTION
**Pull request by JochemBoersma**
I've done a tiny change in docstrings, to make documentation in line with the actual behavior of the function.

* Issue: gh-107028

----
:books: Documentation preview :books:: https://cpython-previews--107029.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/logging.handlers.rst [hunk 1]
```diff
@@ -917,8 +917,7 @@ should, then :meth:`flush` is expected to do the flushing.
 
    .. method:: flush()
 
-      You can override this to implement custom flushing behavior. This version
-      just zaps the buffer to empty.
+      For a :class:`BufferHandler`, flushing means that it zaps the buffer to empty. This method can be overwritten to implement more useful flushing behavior.
```
Comment (evildmp) [inline Doc/library/logging.handlers.rst]: Check line lengths. 

Also, perhaps "empties the buffer" or "clears the buffer".  It's less obvious what "zaps" means.

File: Doc/library/logging.handlers.rst [hunk 2]
```diff
@@ -952,7 +951,7 @@ should, then :meth:`flush` is expected to do the flushing.
 
       For a :class:`MemoryHandler`, flushing means just sending the buffered
       records to the target, if there is one. The buffer is also cleared when
-      this happens. Override if you want different behavior.
+      buffered records are send. Override if you want different behavior.
```
Comment (evildmp) [inline Doc/library/logging.handlers.rst]: "... are sent"

File: Doc/library/logging.handlers.rst (hunk 1 above)
Comment (vsajip) [inline Doc/library/logging.handlers.rst]:
```suggestion
      For a :class:`BufferingHandler` instance, flushing means that it sets the buffer to an empty list. This method can be overwritten to implement more useful flushing behavior.
```

File: Lib/logging/handlers.py [hunk 3]
```diff
@@ -1399,7 +1399,7 @@ def flush(self):
         records to the target, if there is one. Override if you want
         different behaviour.
 
-        The record buffer is also cleared by this operation.
+        The record buffer is only cleared afterwards, when target was set.
```
Comment (evildmp) [inline Lib/logging/handlers.py]: I think I would express this as:  "The record buffer is only cleared if a target has been set."

File: Doc/library/logging.handlers.rst [hunk 4]
```diff
@@ -952,7 +951,7 @@ should, then :meth:`flush` is expected to do the flushing.
 
       For a :class:`MemoryHandler`, flushing means just sending the buffered
```
Comment (vsajip) [inline Doc/library/logging.handlers.rst]:
```suggestion
      For a :class:`MemoryHandler` instance, flushing means just sending the buffered
```

File: Doc/library/logging.handlers.rst (hunk 2 above)
Comment (vsajip) [inline Doc/library/logging.handlers.rst]:
```suggestion
      buffered records are sent to the target. Override if you want different behavior.
```

**Review by vsajip — CHANGES_REQUESTED** [review]
I've suggested a few tweaks.

**Review by evildmp — CHANGES_REQUESTED** [review]

**Comment by vsajip** [comment]
Hi @evildmp , I think our reviews crossed. Hopefully @JochemBoersma can use the best of both sets of suggestions.

File: Doc/library/logging.handlers.rst (hunk 1 above)
Comment (JochemBoersma) [reply Doc/library/logging.handlers.rst]: True, thanks! I've also saw the wrong word for the class. I commit the suggestions.

**Comment by JochemBoersma** [comment]
> Hi @evildmp , I think our reviews crossed. Hopefully @JochemBoersma can use the best of both sets of suggestions.

Hi @vsajip . Thanks for review. To be honest, and for your information: for me the first time to contribute to a project. 
We are having sprint sessions now at the [EuroPython](https://ep2023.europython.eu/sprints).

**Review by evildmp — APPROVED** [review]
This looks good to me. (It can always be further improved of course.)

**Comment by JochemBoersma** [comment]
I have made the requested changes; please review again

**Review by vsajip — APPROVED** [review]

**Pull request merged**